File: 283-move-zeroes/move-zeroes.py
class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        l = 0
        i=0
        for i,v in enumerate(nums):
            if v!=0:
                nums[l],nums[i]=nums[i],nums[l]
                l+=1

        return nums        
            


        

File: 283-move-zeroes/move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int start = 0;
        int end = 0;

        while(end < nums.length){
            if(nums[end] == 0){
                end++;
            } else{
                int temp = nums[end];
                nums[end] = nums[start];
                nums[start] = temp;
                start++;
                end++;
            }
        }
        
    }
}